lang/java/reef-tests/src/test/java/org/apache/reef/tests/yarnnodelabel/NodeLabelTestDriver.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.hadoop.yarn.exceptions.YarnException;
+import org.apache.reef.driver.evaluator.AllocatedEvaluator;
+import org.apache.reef.driver.evaluator.EvaluatorRequest;
+import org.apache.reef.driver.evaluator.EvaluatorRequestor;
+import org.apache.reef.driver.task.TaskConfiguration;
+import org.apache.reef.tang.Configuration;
+import org.apache.reef.tang.annotations.Unit;
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+import org.apache.reef.wake.time.event.StopTime;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Unit
+final class NodeLabelTestDriver {
+  private static final Logger LOG = Logger.getLogger(NodeLabelTestDriver.class.getName());
+
+  private final EvaluatorRequestor evaluatorRequestor;
+
+  private final String nodeLabelExpression = "mylabel";
+  private int labeledContainerCount;
+  private int defaultContainerCount;
+
+  @Inject
+  NodeLabelTestDriver(final EvaluatorRequestor evaluatorRequestor) {
+    this.evaluatorRequestor = evaluatorRequestor;
+    this.labeledContainerCount = 0;
+    this.defaultContainerCount = 0;
+  }
+
+  final class StartHandler implements EventHandler<StartTime> {
+    @Override
+    public void onNext(final StartTime startTime) {
+      final EvaluatorRequest reqToMylabel = EvaluatorRequest.newBuilder()
+          .setNumber(1)
+          .setMemory(64)
+          .setNumberOfCores(1)
+          .setNodeLabelExpression(NodeLabelTestDriver.this.nodeLabelExpression)
+          .build();
+      LOG.log(Level.INFO, "Requested Evaluator with node label: " + reqToMylabel.getNodeLabelExpression());
+      NodeLabelTestDriver.this.evaluatorRequestor.submit(reqToMylabel);
+
+      final EvaluatorRequest reqToDefault = EvaluatorRequest.newBuilder()
+          .setNumber(1)
+          .setMemory(64)
+          .setNumberOfCores(1)
+          .build();
+      LOG.log(Level.INFO, "Requested Evaluator with node label: " + reqToMylabel.getNodeLabelExpression());
+      NodeLabelTestDriver.this.evaluatorRequestor.submit(reqToDefault);
+    }
+  }
+
+  final class EvaluatorAllocatedHandler implements EventHandler<AllocatedEvaluator> {
+    @Override
+    public void onNext(final AllocatedEvaluator allocatedEvaluator) {
+      LOG.log(Level.INFO, "Evaluator Allocated: {0}", allocatedEvaluator);
+
+      final String host = allocatedEvaluator.getEvaluatorDescriptor().getNodeDescriptor()
+          .getInetSocketAddress().getHostString();
+      LOG.log(Level.INFO, "Host: " + host);
+
+      final int port = allocatedEvaluator.getEvaluatorDescriptor().getNodeDescriptor()
+          .getInetSocketAddress().getPort();
+      LOG.log(Level.INFO, "Port: " + port);
+
+      final NodeId nodeid = NodeId.newInstance(host, port);
+      LOG.log(Level.INFO, "NodeId: " + nodeid);
+
+      YarnClient client = YarnClient.createYarnClient();
```

Comment:
@scan3 Thanks for the work! I would like to make a comment. As a client using REEF, creating a `YarnClient` using the host + port information to check the node label of the allocated evaluator seems rather cumbersome. Moreover, the client must import an extra library to resolve the node label information this way, and this must be done for each type of resource manager (ex. Mesos) being used. May I suggest adding this code inside REEF, and provide a way of accessing the node label with the `allocatedEvaluator` instance? (ex. include it as a mapping for the node descriptor)